### PR TITLE
Cut 0.24.0

### PR DESCRIPTION
**Pull request by metalmatze**
Cutting my first release. Please review the changelog.

/cc @mxinden @squat @s-urbaniak

### CODE REVIEW

**Comment by metalmatze** [comment]
For some reason I can't generate the updated `Documentation/` on my machine, even with running the same command as travis...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,28 @@
+## 0.24.0 / 2018-10-10
+
+* [CHANGE] Default to Node Exporter v0.16.0 (#1812)
+* [CHANGE] Update to Go 1.11 (#1855)
+* [CHANGE] Default to Prometheus v2.4.3 (#1929) (#1983)
+* [CHANGE] Default to Thanos v0.1.0 (#1954)
+* [CHANGE] Overhaul metrics while adding triggerBy metric for Alertmanager (#1984)
```
Comment (mxinden) [inline CHANGELOG.md]: Can you add a description just below the release heading that describes what users need to change on their end in regard to the metrics changes?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+* [FEATURE] Add multi namespace support (#1813)
+* [FEATURE] Add SHA field to Prometheus, Alertmanager and Thanos for images (#1847) (#1854)
+* [FEATURE] Add configuration for priority class to be assigned to Pods (#1875)
+* [FEATURE] Configure sampleLimit per ServiceMonitor (#1895)
+* [FEATURE] Add additionalPeers to Alertmanager (#1878)
+* [FEATURE] Add podTargetLabels to ServiceMonitors (#1880)
+* [FEATURE] Relabel target name for Pods (#1896)
+* [FEATURE] Allow configuration of relabel_configs per ServiceMonitor (#1879)
+* [FEATURE] Add illegal update reconciliation by deleting StatefulSet (#1931)
+* [ENHANCEMENT] Set sane bash options on shell scripts (#1801)
```
Comment (mxinden) [inline CHANGELOG.md]: This is just build tooling, not user facing, so no need to mention it here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,28 @@
...
+* [ENHANCEMENT] Update troubleshooting.md (#1808)
```
Comment (mxinden) [inline CHANGELOG.md]: I think so far we have not mentioned any docs updates. So can be skipped as well.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,28 @@
...
+* [ENHANCEMENT] Set Thanos cluster and grpc ip from pod.ip (#1836)
+* [ENHANCEMENT] Add multi namespace end-to-end test (#1861)
```
Comment (mxinden) [inline CHANGELOG.md]: Also not user facing.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,28 @@
...
+* [BUGFIX] Add square brackets around pod IPs (#1881)
```
Comment (mxinden) [inline CHANGELOG.md]: `s/Add square brackets around pod IPs/Add square brackets around pod IPs for IPv6 support`

File: CHANGELOG.md (hunk 1 above)
Comment (metalmatze) [reply CHANGELOG.md]: I've done that. Ready to be reviewed. :+1:

**Comment by metalmatze** [comment]
`make clean` before generating fixed the Documentation :blush: 
I'll merge and create the release after lunch, if no more complaints come in.

**Comment by mxinden** [comment]
:+1:

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,38 @@
...
+Alertmanager controller.
+
+For a full description see: https://github.com/coreos/prometheus-operator/pull/1984#issue-221139702
+
+* [CHANGE] Default to Node Exporter v0.16.0 (#1812)
+* [CHANGE] Update to Go 1.11 (#1855)
+* [CHANGE] Default to Prometheus v2.4.3 (#1929) (#1983)
+* [CHANGE] Default to Thanos v0.1.0 (#1954)
+* [CHANGE] Overhaul metrics while adding triggerBy metric for Alertmanager (#1984)
+* [FEATURE] Add multi namespace support (#1813)
```
Comment (squat) [inline CHANGELOG.md]: we should add a note that says that this introduces a breaking change. The operator no longer accepts the `--namespace` flag but rather `--namespaces`

Comment (mxinden) [reply CHANGELOG.md]: @metalmatze can you follow up with a fix to `release-0.24` branch and merge that back into `master` once its merged?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,38 @@
...
+
+The same happened with `prometheus_operator_alertmanager_spec_replicas` and `prometheus_operator_prometheus_spec_replicas`
+which is now called `prometheus_operator_spec_replicas` and also has the "controller" label.
+
+The `prometheus_operator_triggered_total` metric now has a "controller" label as well and finally instruments the
+Alertmanager controller.
+
+For a full description see: https://github.com/coreos/prometheus-operator/pull/1984#issue-221139702
+
+* [CHANGE] Default to Node Exporter v0.16.0 (#1812)
```
Comment (brancz) [inline CHANGELOG.md]: Kube-prometheus is unversioned so these changes should not be in the Prometheus operator changelog.

**Pull request merged**